subprojects/logging/src/main/java/org/gradle/internal/logging/console/WorkInProgressRenderer.java

```diff
@@ -75,23 +77,47 @@ public void onOutput(OutputEvent event) {
             operations.progress(progressEvent.getStatus(), progressEvent.getProgressOperationId());
         } else if (event instanceof EndOutputEvent) {
             progressArea.setVisible(false);
-        } else if (event instanceof MaxWorkerCountChangeEvent) {
-            int newCount = consoleLayoutCalculator.calculateNumWorkersForConsoleDisplay(
-                ((MaxWorkerCountChangeEvent) event).getNewMaxWorkerCount());
-            resizeTo(newCount);
         }
 
         listener.onOutput(event);
     }
 
     @Override
     public void onOutput(Iterable<OutputEvent> events) {
-        super.onOutput(events);
+        Set<OperationIdentifier> completeEventOperationIds = toOperationIdSet(toProgressCompleteEvents(events));
+        Set<OperationIdentifier> startEventOperationIdsToSkip = new HashSet<OperationIdentifier>();
+
+        for (OutputEvent event : events) {
+            if (event instanceof ProgressStartEvent && completeEventOperationIds.contains(((ProgressStartEvent) event).getProgressOperationId())) {
```

Comment:
This achieves one of our other stories for polishing console output. I feel like this is orthogonal to the changes we made recently, though, so I would be surprised if this fixed all the issues we are seeing. 

Let's reconsider if this is really better as a separate change.